gauss_jordan.go

```diff
@@ -129,7 +129,7 @@ func GaussJordan(linearSystem [][]*Int, linearSystemResult []*Int) ([]*Int, erro
 	}
 	for i := 0; i < nrows; i++ {
 		for j := 0; j < ncols; j++ {
-			if extendedMatrix[i][j].Cmp(NewInt(0)) != 0 {
+			if extendedMatrix[i][j].Cmp(NewInt(0, extendedMatrix[i][j].Base)) != 0 {
```

Comment:
Makes sense that we are comparing with the same position in both matrices, but is it even possible to have matrices with different bases in different elements?
if not possible/coherent mathematically, we can avoid accessing the element a second time (or we shoult at least store a pointer to the element so we don't need to access it twice). We can't count on the optmizer here figuring out it's the same access because it may think that Cmp does change the value of the first.

Reply:
so yeah we assume that all matrices have the same base prime for the entire matrix. If you don't instansiate your matrix in this way you will for sure have undesired behavior. IN cmp you only compare the "values" not the bases anyway, so really no matter what base you pass here Cmp will still always work correctly.

Reply:
we could change it to this instead:
``` extendedMatrix[i][j].Value.Cmp(big.NewInt(0))```

Reply:
But you think in the creation of `NewInt(0, extended...)` we can use always the same Base (the first for example), to avoid accessing the others?
Honeslty I think with processor pipelining the optmimization might be worthless, but it's for sure not making it slower.